utils/logger.py

```diff
@@ -60,12 +61,20 @@ def __init__(
         """
         if show_message:
             print(msg)
+        
+        # Load the dotenv configuration in case config class has not been used
+        dotenv.load_dotenv()
 
         self.logger_name = logger_name
         self.log_level = os.getenv("LOG_LEVEL", log_level)
         self.log_level_console = os.getenv("LOG_LEVEL_CONSOLE", self.log_level)
-        self.logfile = logfile
-        self.logfile_maxSize = 1048576 * 100
+        _logfile = os.getenv("LOG_FILE_NAME")
+        if _logfile is not None and len(str(_logfile)) >= 1:
+            # enforce the use of logs/ directory
+            # use last token if user attemps to provide full paths
+            _logfile = "logs/"+str(_logfile).split('/')[-1]
+        self.logfile = _logfile if _logfile else logfile
```

Comment:
what's the actual behavior if `logfile` ends up `None`?

Reply:
It disables logging to file (see line 105)
```python
        if self.logfile is not None:
```

Reply:
Since the "logfile" init arg has a default value, the only way to actually disable logging is to pass `None` to init (and leave the env var unset)?

I think the logfile init arg should default to `None`.  Then you can enable logging by either setting the env var, or passing an arg to init.

also while you don't need to change this here, as i work on config I am going to centralize the config loading.  Having different components reading the env vars directly to determine their own config is not the best structure imho.  Loading the config in a central location makes for a better abstraction as the config format changes, and allows us to do config validation, overriding, etc, in a single spot

Reply:
Okay. I'll address this.

Reply:
Done.